test/parallel/test-http2-update-settings.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// This test ensures that the Http2SecureServer and Http2Server
+// settings are updated when the setting object is valid.
+// When the setting object is invalid, this test ensures that
+// updateSettings throws an exception.
+
+const common = require('../common');
+if (!common.hasCrypto) { common.skip('missing crypto'); }
+const assert = require('assert');
+const http2 = require('http2');
+
+testUpdateSettingsWith({
+  server: http2.createSecureServer(),
+  newServerSettings: {
+    'headerTableSize': 1,
+    'initialWindowSize': 1,
+    'maxConcurrentStreams': 1,
+    'maxHeaderListSize': 1,
+    'maxFrameSize': 16385,
+    'enablePush': false,
+    'enableConnectProtocol': true
+  }
+});
+testUpdateSettingsWith({
+  server: http2.createServer(),
+  newServerSettings: {
+    'enablePush': false
+  }
+});
+
+function testUpdateSettingsWith({ server, newServerSettings }) {
+  const faultySettingException = {
+    message: 'The "settings" argument must be of type object. ' +
+    'Received type string (\'\')',
+    code: 'ERR_INVALID_ARG_TYPE',
+    name: 'TypeError'
+  };
+  const faultyValueException = {
+    message: 'Invalid value for setting "maxHeaderListSize": foo',
+    code: 'ERR_HTTP2_INVALID_SETTING_VALUE',
+    name: 'RangeError'
+  };
+  const faultySettingValue = {
+    'maxHeaderListSize': 'foo'
+  };
+  const oldServerSettings = getServerSettings(server);
+  assert.notDeepStrictEqual(oldServerSettings, newServerSettings);
+  server.updateSettings(newServerSettings);
+  const updatedServerSettings = getServerSettings(server);
+  assert.deepStrictEqual(updatedServerSettings, { ...oldServerSettings,
+                                                  ...newServerSettings });
+  assert.throws(() => server.updateSettings(''),
+                faultySettingException);
+  assert.throws(() => server.updateSettings(faultySettingValue),
+                faultyValueException);
```

Comment:
Nit: faultySettingException and faultyValueException can probably be inlined since they are only used once, it will also improve coupling/understanding of the code IMO.

Reply:
You are right, inlining the objects makes it easier to read the tests. I kept the vars from my logic before I refactored it into a single function